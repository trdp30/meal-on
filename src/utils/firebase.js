// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAnalytics } from "firebase/analytics";
import { getAuth } from "firebase/auth";

const firebaseConfig = {
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_FIREBASE_APP_ID,
  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENT_ID,
};

let app, analytics, auth;

// Initialize Firebase
// export default async function initializeFirebase() {
//   app = await initializeApp(firebaseConfig, "menuOnRoad");
//   analytics = await getAnalytics(app);

//   if (process.env.NODE_ENV !== "production") {
//     // analytics.app.disable();
//   }
//   auth = await getAuth(app);
//   return {
//     app,
//     analytics,
//     auth,
//   };
// }

app = initializeApp(firebaseConfig, "menuOnRoad");
analytics = getAnalytics(app);

if (process.env.NODE_ENV !== "production") {
  // analytics.app.disable();
}
auth = getAuth(app);

console.log("auth", auth);

export { app, analytics, auth };

export default app;
