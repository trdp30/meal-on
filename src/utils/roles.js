const roles = { admin: "ADMIN", owner: "OWNER", manager: "MANAGER" };

export default roles;
