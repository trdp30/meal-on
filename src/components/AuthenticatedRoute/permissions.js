import { ROLE_MASTER_RECRUITER, ROLE_RECRUITER } from 'utils/data/roles';

export const DEFAULT_ROUTE_PERMISSIONS = ['/login', '/logout', '/dashboard'];
export const ROUTE_PERMISSIONS = {
  [ROLE_MASTER_RECRUITER]: [
    '/dashboard',
    '/workflow',
    '/workflow/:workflow_id',
    '/workflow/:workflow_id/invite/:step_id',
    '/workflow/:workflow_id/drive/:drive_id',
  ],
  [ROLE_RECRUITER]: [
    '/dashboard',
    '/workflow',
    '/workflow/:workflow_id',
    '/workflow/:workflow_id/invite/:step_id',
    '/workflow/:workflow_id/drive/:drive_id',
  ],
  // [ROLE_ADMIN]: ['/question-bank', '/taxonomy', '/library', '/admin'],
  // [ROLE_CONTENT_MANAGER]: ['/question-bank', '/taxonomy', '/library'],
};
