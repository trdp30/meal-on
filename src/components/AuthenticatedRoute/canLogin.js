import { compact, flatMap, memoize } from 'lodash';
import { DEFAULT_ROUTE_PERMISSIONS, ROUTE_PERMISSIONS } from './permissions';

export const canLogin = ({ roles, path, isAnonymous }) => {
  const allowed = compact(
    flatMap(ROUTE_PERMISSIONS, (routes, role) => (roles && roles.includes(role) ? routes : null)).concat(
      DEFAULT_ROUTE_PERMISSIONS,
    ),
  );

  return !isAnonymous && allowed.includes(path);
};

export default memoize(canLogin);
