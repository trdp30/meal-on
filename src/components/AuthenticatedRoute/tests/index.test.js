/**
 *
 * Tests for AuthenticatedRoute
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'utils/testHelpers';
import '@testing-library/jest-dom'; // add some helpful assertions

import { Route, Routes } from 'react-router-dom';
import AuthenticatedRoute from '../index';

describe('<AuthenticatedRoute />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<AuthenticatedRoute />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Redirect to login on logged out', () => {
    const { getByText } = render(
      <Routes>
        <Route
          path="/dashboard"
          element={
            <AuthenticatedRoute path="/dashboard">
              <div>dashboard</div>
            </AuthenticatedRoute>
          }
        />
        <Route path="/login" element={<div>Login</div>} />
      </Routes>,
      {
        user: { isAnonymous: true },
        route: ['/dashboard'],
      },
    );
    expect(getByText('Login')).toBeTruthy();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<AuthenticatedRoute />);
    expect(container).toMatchSnapshot();
  });
});
