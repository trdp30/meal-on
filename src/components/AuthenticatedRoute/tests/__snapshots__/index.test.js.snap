// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticatedRoute /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="h-full bg-slate-50 relative"
  >
    <div
      class="h-full flex justify-center items-center "
    >
      <div
        class="flex shadow-lg items-center p-6 rounded-md bg-white"
      >
        <div
          class="flex flex-col space-y-8 flex-shrink-0 w-full md:w-1/2"
        >
          <h2
            class="flex md:text-5xl text-2xl font-semibold leading-8 align-middle items-center"
          >
            <svg
              aria-hidden="true"
              class="w-7 h-7 text-red-600 inline md:hidden"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M13.477 14.89A6 6 0 015.11 6.524l8.367 8.368zm1.414-1.414L6.524 5.11a6 6 0 018.367 8.367zM18 10a8 8 0 11-16 0 8 8 0 0116 0z"
                fill-rule="evenodd"
              />
            </svg>
             
            Forbidden!
          </h2>
          <p
            class="text-lg break-word"
          >
            We are Sorry
            ..
          </p>
          <p
            class="text-md break-word"
          >
            The page you're trying to access has restricted access.
          </p>
          <div
            class="w-[120px]"
          >
            <button
              class="text-black-700  border border-gray-200 hover:border-blue-400 hover:text-blue-700 active:border-blue-300 focus:ring-blue-300 focus:ring-2 active:text-blue-700 active:bg-blue-50 focus-visible:outline-0 px-4 py-2.5 text-sm font-medium leading-4 rounded-lg w-full flex justify-center relative"
              type="button"
            >
              <span
                class="mr-2.5"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-house "
                  data-icon="house"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <span>
                Go Home
              </span>
            </button>
          </div>
        </div>
        <img
          alt="Forbidden"
          class="md:max-w-md md:w-1/2 hidden md:block -mr-20"
          src="IMAGE_MOCK"
        />
      </div>
    </div>
  </div>
</div>
`;
